substrate/client/network/src/request_responses.rs

```diff
@@ -64,7 +66,93 @@ use std::{
 	time::{Duration, Instant},
 };
 
-pub use libp2p::request_response::{Config, InboundFailure, OutboundFailure, RequestId};
+pub use libp2p::request_response::{Config, RequestId};
+
+/// Possible failures occurring in the context of sending an outbound request and receiving the
+/// response.
+
+#[derive(Debug, thiserror::Error)]
+pub enum OutboundFailure {
+	/// The request could not be sent because a dialing attempt failed.
+	DialFailure,
+	/// The request timed out before a response was received.
+	Timeout,
+	/// The connection closed before a response was received.
+	ConnectionClosed,
+	/// The remote supports none of the requested protocols.
+	UnsupportedProtocols,
+}
+
+impl Display for OutboundFailure {
+	fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+		match self {
+			OutboundFailure::DialFailure => write!(f, "Failed to dial the requested peer"),
```

Comment:
nit: Could we use `thiserror` instead? 

```rust
pub enum OutbounFailure {
   #[error("Failed to dial the requested peer")]
   DialFailure,
```

Reply:
True! Sorry, @dmitry-markin mentioned the same thing, but I misunderstood his comment!